bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHAApplianceHandler.java

```diff
@@ -0,0 +1,497 @@
+                return;
+            }
+
+            try {
+                PlugwiseHAController controller = bridgeHandler.getController();
+                if (controller != null) {
+                    this.appliance = getEntity(controller, true);
+                    Appliance localAppliance = this.appliance;
+                    if (localAppliance != null) {
+                        if (localAppliance.isBatteryOperated()) {
+                            addBatteryChannels();
+                        }
+                        setApplianceProperties();
+                        updateStatus(ONLINE);
+                    } else {
+                        updateStatus(OFFLINE);
+                    }
+                } else {
+                    updateStatus(OFFLINE, BRIDGE_OFFLINE);
+                }
+            } catch (PlugwiseHAException e) {
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    protected @Nullable Appliance getEntity(PlugwiseHAController controller, Boolean forceRefresh)
+            throws PlugwiseHAException {
+        PlugwiseHAThingConfig config = getPlugwiseThingConfig();
+        Appliance appliance = controller.getAppliance(config.getId(), forceRefresh);
+
+        return appliance;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void handleCommand(Appliance entity, ChannelUID channelUID, Command command) throws PlugwiseHAException {
+        String channelID = channelUID.getIdWithoutGroup();
+
+        PlugwiseHABridgeHandler bridge = this.getPlugwiseHABridge();
+        if (bridge == null) {
+            return;
+        }
+
+        PlugwiseHAController controller = bridge.getController();
+        if (controller == null) {
+            return;
+        }
+
+        switch (channelID) {
+            case APPLIANCE_LOCK_CHANNEL:
+                if (command instanceof OnOffType) {
+                    try {
+                        if (command == OnOffType.ON) {
+                            controller.switchRelayLockOn(entity);
+                        } else {
+                            controller.switchRelayLockOff(entity);
+                        }
+                    } catch (PlugwiseHAException e) {
+                        logger.warn("Unable to switch relay lock {} for appliance '{}'", (State) command,
+                                entity.getName());
+                    }
+                }
+                break;
+            case APPLIANCE_OFFSET_CHANNEL:
+                if (command instanceof QuantityType) {
+                    Unit<Temperature> unit = entity.getOffsetTemperatureUnit().orElse(UNIT_CELSIUS).equals(UNIT_CELSIUS)
+                            ? SIUnits.CELSIUS
+                            : ImperialUnits.FAHRENHEIT;
+                    QuantityType<Temperature> state = ((QuantityType<Temperature>) command).toUnit(unit);
```

Comment:
You could use the wildcard, then the warning suppression can be removed.
```suggestion
                    QuantityType<?> state = ((QuantityType<?>) command).toUnit(unit);
```

Reply:
Fixed